### PR TITLE
fix: needs error with context that includes slash

### PR DESCRIPTION
**Pull request by sergeief**
Fix an issue that causes `needs:` dependency errors when using kube context with slashes in it. This works by joining all components of ID except last two (`namespace` and `name`).

Assumes `namespace` and `name` cannot contain slashes.

Fixes https://github.com/helmfile/helmfile/issues/203

### CODE REVIEW

File: pkg/app/diff_test.go [hunk 1]
```diff
@@ -1291,15 +1317,20 @@ changing working directory back to "/path/to"
 					t.Errorf("unexpected error creating vals runtime: %v", err)
 				}
 
+				overrideKubeContext := tc.overrideContext
+				if tc.overrideContext == "" {
+					overrideKubeContext = "default"
+				}
+
```
Comment (yxxhero) [inline pkg/app/diff_test.go:1324]:
```go
				overrideKubeContext := "default"
				if tc.overrideContext != "" {
					overrideKubeContext = tc.overrideContext
				}
```

**Comment by yxxhero** [comment]
@sergeief Can you add a testcase like the issue: https://github.com/helmfile/helmfile/issues/203 Thanks very much.

**Comment by sergeief** [comment]
@yxxhero thanks, added

**Review by yxxhero — APPROVED** [review]
LGTM

**Pull request merged**